2019 2020 training

bot190 opened the pull request:
This is a work in progress training plan for the 2019-2020 season. The goal is to teach the required Computer Science concepts and Java syntax in the context of programming an FRC robot. This is intended to better prepare students for programming advanced robot functionality during build season. 

When we get closer to the start of training and the various libraries we use have settled down I will add a folder with initial robot code. This folder will be used throughout the training as we build up functionality. 

This is very much a work in progress and all comments are welcome.

bot190 committed:
Add Training Plan Documents

Adds Initial list of training requirements.
Also includes task list for intermediate students self guided training. 

This is a work in progress.

bot190 committed:
Initial training plan for Beginners

Adds a rough training plan for the first 5 software training meetings. This is written with a focus on robot code, while reinforcing concepts using more traditional console programs where necessary.

Rohan-Sreenivasan commented:
Thank you, this looks really good. I have some feedback that we could talk about in the meeting.
Thanks,Rohan Sreenivasan
    On Sunday, August 18, 2019, 09:10:15 PM EDT, Ben Beauregard <notifications@github.com> wrote:  
 
 
This is a work in progress training plan for the 2019-2020 season. The goal is to teach the required Computer Science concepts and Java syntax in the context of programming an FRC robot. This is intended to better prepare students for programming advanced robot functionality during build season.

When we get closer to the start of training and the various libraries we use have settled down I will add a folder with initial robot code. This folder will be used throughout the training as we build up functionality.

This is very much a work in progress and all comments are welcome.

You can view, comment on, or merge this pull request online at:

  https://github.com/chopshop-166/SoftwareTraining/pull/2

Commit Summary
   
   - Add Training Plan Documents
   - Initial training plan for Beginners

File Changes
   
   -  A Training Plan/BeginningTraining.md (133) 
   -  A Training Plan/IntermediateTraining.md (26) 
   -  A Training Plan/Training.md (50) 

Patch Links:
   
   - https://github.com/chopshop-166/SoftwareTraining/pull/2.patch
   - https://github.com/chopshop-166/SoftwareTraining/pull/2.diff

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub, or mute the thread.

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+  - [Git](https://git-scm.com/)
+  - [Radio programming tool](https://docs.wpilib.org/en/latest/docs/software/getting-started/radio-programming.html) (Advanced Students)
+ - [ ] Demo of basic driving robot code
+  - Start with OI and subsystem created, including default command defined
```
In commands-experimental (in the namespace `edu.wpi.first.wpilibj.frc2.command`), default commands are optionally set in the constructor, instead of having to override a particular method. I assume we plan on teaching the new library?

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+  - Add single line comment inside drive command
+  - Add multi-line function comment for drive command
+ - [ ] Variables
+  - int
```
Please use backticks around code in our markdown.

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+ - [ ] Variables
+  - int
+    - Whole numbers, integars are passed to create motor controllers
+  - float/double
```
Do we even use floats anywhere? I mean we might as well talk about them, but a quick "These are like doubles but smaller and less precise"

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+  - int
+    - Whole numbers, integars are passed to create motor controllers
+  - float/double
+    - Fractional numbers
```
Floating-point. While it's less intuitive than "fractional", it's also a lot more accurate.

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+    - Explore printing other things
+  - Basic input
+    - Ask for users name, greet them
+    - Ask users age, print "You can drive" / "You can't drive" as appropriate
```
As a bonus challenge they can do "You can drive with supervision" for age 15.5

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+  - Cover syntax
+    - `public class Name {}`
+    - Constructor, has the same name as the class
+    - Where do Member variables go?
```
Make sure we tell them that they should go up at the top of the class - it's annoying trying to find variables and constants that are mixed in with the methods.

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+    - `talk` overloaded to "bark"
+  - Example 'Cat' extends 'Pet'
+    - `talk` overloaded to "meow"
+  - 
```
Random bullet here

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,133 @@
+  - Example 'Dog' extends 'Pet'
+    - `talk` overloaded to "bark"
+  - Example 'Cat' extends 'Pet'
+    - `talk` overloaded to "meow"
```
Cats are evil

bot190 committed:
Add Sixth Meeting agenda

Sixth meeting will continue the discussion of Classes, and describe
interfaces.

Also added a few more rainy day topics.

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,152 @@
+    - Print average of numbers gathered above
+  - `while` loop syntax
+    - Simple guessing game, ask user for number until it matches
+    - Explain infinite loop as basis of robot control,
```
Trailing comma

msoucy commented on Training Plan/BeginningTraining.md:
```diff
@@ -0,0 +1,152 @@
+  - Used to add information about code inside the code
+  - They can be extended with custom behavior
+  - Explore the usage of the `RobotMapFor("MAC ADDRESS")` annotation in the robot map
+ - [ ] [Collections](https://docs.oracle.com/javase/8/docs/api/?java/util/Collections.html)
```
Use the Java 11 javadocs - that's the version we support.

bot190 committed:
Fix review comments

This updates the Java collections documentation link, and removes an extra comma.

bot190 committed:
Add Training Project

This adds a basic command based project that will be expanded throughout training as a learning tool.

msoucy commented:
LGTM :shipit:
